File: Examples/VBasic/GroupDocs.Signature.Examples.VBasic/Azure/AzureDataHandler.vb
Imports Microsoft.WindowsAzure.Storage.Auth
Imports Microsoft.WindowsAzure.Storage.Blob
Imports Microsoft.WindowsAzure.Storage

'ExStart:azuredatahandling
Public MustInherit Class AzureDataHandler
    Private _remoteStorage As CloudBlobClient
    Private _containerName As String

    ''' <summary>
    ''' Handles azure data
    ''' </summary>
    ''' <param name="endpoint"></param>
    ''' <param name="accountName"></param>
    ''' <param name="accountKey"></param>
    ''' <param name="containerName"></param>
    Public Sub New(endpoint As String, accountName As String, accountKey As String, containerName As String)
        Dim credentials = New StorageCredentials(accountName, accountKey)
        Dim account As New CloudStorageAccount(credentials, New Uri(endpoint), Nothing, Nothing, Nothing)
        _remoteStorage = account.CreateCloudBlobClient()
        _containerName = containerName
        Dim defTimeout As System.Nullable(Of TimeSpan) = _remoteStorage.DefaultRequestOptions.ServerTimeout
        _remoteStorage.DefaultRequestOptions.ServerTimeout = TimeSpan.FromSeconds(10)
        Dim container As CloudBlobContainer = _remoteStorage.GetContainerReference(containerName)
        container.CreateIfNotExists()
        _remoteStorage.DefaultRequestOptions.ServerTimeout = defTimeout
    End Sub

    ''' <summary>
    ''' Gets container reference
    ''' </summary>
    ''' <returns>Returns container</returns>
    Public Function GetContainerReference() As CloudBlobContainer
        Dim container As CloudBlobContainer = _remoteStorage.GetContainerReference(_containerName)
        Return container
    End Function
End Class
'ExEnd:azuredatahandling

File: Examples/CSharp/GroupDocs.Signature.Examples.CSharp/Azure/AzureDataHandler.cs
using Microsoft.WindowsAzure.Storage;
using Microsoft.WindowsAzure.Storage.Auth;
using Microsoft.WindowsAzure.Storage.Blob;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GroupDocs.Signature.Examples.CSharp
{
    //ExStart:azuredatahandling
    public abstract class AzureDataHandler
    {
        private CloudBlobClient _remoteStorage;
        private string _containerName;
       
        /// <summary>
        /// Handles azure data
        /// </summary>
        /// <param name="endpoint"></param>
        /// <param name="accountName"></param>
        /// <param name="accountKey"></param>
        /// <param name="containerName"></param>
        public AzureDataHandler(string endpoint,
            string accountName,
            string accountKey,
            string containerName)
        {
            var credentials = new StorageCredentials(accountName, accountKey);
            CloudStorageAccount account = new CloudStorageAccount(credentials, new Uri(endpoint), null, null, null);
            _remoteStorage = account.CreateCloudBlobClient();
            _containerName = containerName;
            TimeSpan? defTimeout = _remoteStorage.DefaultRequestOptions.ServerTimeout;
            _remoteStorage.DefaultRequestOptions.ServerTimeout = TimeSpan.FromSeconds(10);
            CloudBlobContainer container = _remoteStorage.GetContainerReference(containerName);
            container.CreateIfNotExists();
            _remoteStorage.DefaultRequestOptions.ServerTimeout = defTimeout;
        }

        /// <summary>
        /// Gets container reference
        /// </summary>
        /// <returns>Returns container</returns>
        public CloudBlobContainer GetContainerReference()
        {
            CloudBlobContainer container = _remoteStorage.GetContainerReference(_containerName);
            return container;
        }
    }
    //ExEnd:azuredatahandling
}
